llvm/test/CodeGen/RISCV/rvv/fixed-vectors-strided-load-store-asm.ll

```diff
@@ -144,9 +144,9 @@ define void @gather_zero_stride(ptr noalias nocapture %A, ptr noalias nocapture
 ; CHECK-NEXT:    vsetvli zero, a3, e8, m1, ta, ma
 ; CHECK-NEXT:  .LBB3_1: # %vector.body
 ; CHECK-NEXT:    # =>This Inner Loop Header: Depth=1
-; CHECK-NEXT:    lbu a3, 0(a1)
-; CHECK-NEXT:    vle8.v v8, (a0)
-; CHECK-NEXT:    vadd.vx v8, v8, a3
+; CHECK-NEXT:    vlse8.v v8, (a1), zero
```

Comment:
Which PR will fix this?

Reply:
https://github.com/llvm/llvm-project/pull/97798, but I believe @BeMg is working on a CodeGenPrepare version which should handle more cases

Reply:
I messed up who is working on it. It’s @yetingk

Reply:
In #97798 I had to make sure to use build_vector for fixed-length vectors and splat_vector for scalable vectors.

For CodeGenPrepare does using `IRBuilder::CreateVectorSplat` help? I presume that would get lowered to build_vector and splat_vector for fixed-length and scalable vectors respectively.

Reply:
Oh I see, but we want the vmv.v.x/vmv.v.f to have vl=evl? Does wrapping the splat in a `@llvm.vp.select` with the passthru and false both set to undef, but using the original mask and evl work? #97989 recently landed which may help set vl=evl

Reply:
And we can't use `@llvm.riscv.vmv.v.x` because that's only on scalable vectors right?

I think it should still be correct if we splat with AVL=VLMAX because the disabled lanes are poison anyway. It's a bit annoying that there's no good way of controlling the AVL here though but we could leave that to later?